src/pixel/url-collector.ts

```diff
@@ -0,0 +1,51 @@
+import { UrlCollectionModes } from '../model/url-collection-mode'
+import { LiveConnectConfig } from '../types'
+import { ParsedUrl, urlParamsArray } from '../utils/url'
+
+export function collectUrl(urlString: string, conf: LiveConnectConfig): string {
+  const url = new ParsedUrl(urlString)
+  if (conf.urlCollectionMode === UrlCollectionModes.noPageUrl) {
+    url.pathname = '/'
+  }
+  url.search = paramsToKeep(url, conf)
+  return url.toString()
+}
+
+export function blockedQueryParams(urlString: string | undefined, conf: LiveConnectConfig): string[] {
+  if (!urlString) {
+    return []
+  }
+  const url = new ParsedUrl(urlString)
+  return paramsToDelete(url, conf)
+}
+
+export function isPageRemoved(urlString: string | undefined, conf: LiveConnectConfig): boolean {
+  if (!urlString || conf.urlCollectionMode !== UrlCollectionModes.noPageUrl) {
+    return false
+  }
+  const url = new ParsedUrl(urlString)
```

Comment:
Why do we not trust our configuration?

Reply:
In the state model the url is optional so it can be undefined.

Regarding the `url.pathname.length > 1 ` we want to have a field that tracks when the path is actually removed from the url, not just when the "mode" is `no_page` and since the initial `/` is considered part of the path we need to check if the length is `> 1`.

Just as reference, I named the things in the same way as the js [URL class](https://developer.mozilla.org/en-US/docs/Web/API/URL/URL), which I was using initially, but then I found out it wasn't supported on IE.

Reply:
that makes sense, thanks 👍